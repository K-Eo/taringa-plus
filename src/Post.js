import React from 'react'
import TimeAgo from 'react-timeago'
import { Card, Image } from 'semantic-ui-react'

import { getPostImageURL, esFormatter } from './Utils'

const Post = (props) => (
  <Card key={props.id}>
    <Image src={getPostImageURL(props)}/>
    <Card.Content>
      <Card.Header as='a' href={props.canonical} target='_blank'>
        {props.title}
      </Card.Header>
      <Card.Meta>
        <TimeAgo
          date={props.created}
          formatter={esFormatter}
        />
      </Card.Meta>
    </Card.Content>
    <Card.Content extra>
      <Image src={props.owner.avatar.medium} avatar/>
      {props.owner.nick}
    </Card.Content>
  </Card>
)

export default Post
