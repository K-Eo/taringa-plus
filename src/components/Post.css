.animated-background {
  animation-duration: 1s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: placeholderShimmer;
  animation-timing-function: linear;
  background: #f6f7f8;
  background: linear-gradient(to right, #eee 8%, #ddd, 18%, #eee 33%);
  background-size: 800px 104px;
  height: 362.95px;
  position: relative;
}

.ph-post-img {
  height: 199.375px;
}

.bg-masker.top-separator {
  height: 14px;
  left: 0;
  right: 0;
  top: 199.375px;
}

.bg-masker.left-separator {
  border-bottom-left-radius: 0.28571429rem;
  bottom: 0;
  left: 0;
  top: 199.375px;
  width: 14px;
}

.bg-masker.right-separator {
  border-bottom-right-radius: 0.28571429rem;
  bottom: 0;
  right: 0;
  top: 199.375px;
  width: 14px;
}

.bg-masker.mid-separator {
  height: 8px;
  left: 0;
  right: 0;
  top: 256.375px;
}

.bg-masker.date-separator {
  height: 18px;
  right: 0;
  bottom: 81px;
  width: 140px;
}

.bg-masker.extra-separator {
  height: 32px;
  left: 0;
  right: 0;
  bottom: 49px;
}

.bg-masker.bottom-separator {
  border-bottom-left-radius: 0.38571429rem;
  border-bottom-right-radius: 0.38571429rem;
  bottom: 0;
  height: 10.5px;
  left: 0;
  right: 0;
}
