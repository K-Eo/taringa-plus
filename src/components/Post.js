import React from 'react'
import TimeAgo from 'react-timeago'
import { Card, Image } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

import { getPostImageURL, esFormatter } from '../Utils'

const Post = (props) => (
  <Card key={props.id}>
    <Image src={getPostImageURL(props)}/>
    <Card.Content>
      <Card.Header
        as={Link}
        to={`/post/${props.id}`}
      >
        {props.title}
      </Card.Header>
      <Card.Meta>
        <TimeAgo
          date={props.created}
          formatter={esFormatter}
        />
      </Card.Meta>
    </Card.Content>
    <Card.Content extra>
      <Image src={props.owner.avatar.medium} avatar/>
      {props.owner.nick}
    </Card.Content>
  </Card>
)

export default Post
