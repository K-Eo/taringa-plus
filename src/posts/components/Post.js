import React from 'react'
import TimeAgo from 'react-timeago'
import { Card, Image } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

import './Post.css'
import { getPostImageURL, esFormatter } from '../../Utils'

const Post = (props) => {
  if (props.placeholder) {
    return (
      <Card>
        <div className='animated-background'>
          <div className='ph-post-img'/>
          <div className='bg-masker top-separator'/>
          <div className='bg-masker left-separator'/>
          <div className='bg-masker right-separator'/>
          <div className='bg-masker mid-separator'/>
          <div className='bg-masker date-separator'/>
          <div className='bg-masker extra-separator'/>
          <div className='bg-masker bottom-separator'/>
        </div>
      </Card>
    )
  }

  return (
    <Card>
      <Image src={getPostImageURL(props)} style={{ height: 199.8 }}/>
      <Card.Content>
        <Card.Header
          as={Link}
          to={`/post/${props.id}`}
        >
          {props.title}
        </Card.Header>
        <Card.Meta>
          <TimeAgo
            date={props.created}
            formatter={esFormatter}
          />
        </Card.Meta>
      </Card.Content>
      <Card.Content extra>
        <Image src={props.owner.avatar.medium} avatar/>
        {props.owner.nick}
      </Card.Content>
    </Card>
  )
}

export default Post
