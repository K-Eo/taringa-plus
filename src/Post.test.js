import React from 'react'
import Post from './Post'
import renderer from 'react-test-renderer'

it('Renders Post', () => {
  const now = Date.now('Dec 21, 2017')
  Date.now = jest.genMockFunction().mockReturnValue(now)

  const item = {
    id: 1,
    created: new Date('Dec 20, 2017'),
    canonical: 'canonical-url',
    title: 'post-title',
    owner: {
      avatar: {
        medium: 'medium-avatar'
      },
      nick: 'owner-nick'
    }
  }

  const component = renderer.create(
    <Post {...item}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
