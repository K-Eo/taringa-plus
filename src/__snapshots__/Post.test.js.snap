// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Post 1`] = `
<div
  className="ui card"
  href={undefined}
  onClick={[Function]}
>
  <img
    className="ui image"
    src=""
  />
  <div
    className="content"
  >
    <a
      className="header"
      href="canonical-url"
      target="_blank"
    >
      post-title
    </a>
    <div
      className="meta"
    >
      <time
        dateTime="2017-12-21T06:00:00.000Z"
        title="Dec 21, 2017"
      >
        hace 1 d√≠a
      </time>
    </div>
  </div>
  <div
    className="extra content"
  >
    <img
      className="ui avatar image"
      src="medium-avatar"
    />
    owner-nick
  </div>
</div>
`;
